vault/identity_store_oidc_provider.go

```diff
@@ -161,3 +217,106 @@ func (i *IdentityStore) pathOIDCAssignmentExistenceCheck(ctx context.Context, re
 
 	return entry != nil, nil
 }
+
+// pathOIDCCreateUpdateProvider is used to create a new named provider or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateProvider(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+
+	var provider namedProvider
+	if req.Operation == logical.UpdateOperation {
+		entry, err := req.Storage.Get(ctx, namedProviderPath+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(&provider); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if issuerRaw, ok := d.GetOk("issuer"); ok {
+		provider.Issuer = issuerRaw.(string)
+	} else if req.Operation == logical.CreateOperation {
+		provider.Issuer = d.Get("issuer").(string)
+	}
+
+	if allowedClientIDsRaw, ok := d.GetOk("allowed_client_ids"); ok {
+		provider.AllowedClientIDs = allowedClientIDsRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		provider.AllowedClientIDs = d.Get("allowed_client_ids").([]string)
+	}
+
+	if scopesRaw, ok := d.GetOk("scopes"); ok {
+		provider.Scopes = scopesRaw.([]string)
+	} else if req.Operation == logical.CreateOperation {
+		provider.Scopes = d.Get("scopes").([]string)
+	}
+
+	// store named provider
+	entry, err := logical.StorageEntryJSON(namedProviderPath+name, provider)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := req.Storage.Put(ctx, entry); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// pathOIDCListProvider is used to list named providers
+func (i *IdentityStore) pathOIDCListProvider(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	providers, err := req.Storage.List(ctx, namedProviderPath)
+	if err != nil {
+		return nil, err
+	}
+	return logical.ListResponse(providers), nil
+}
+
+// pathOIDCReadProvider is used to read an existing provider
+func (i *IdentityStore) pathOIDCReadProvider(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+
+	entry, err := req.Storage.Get(ctx, namedProviderPath+name)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	var storedNameProvider namedProvider
+	if err := entry.DecodeJSON(&storedNameProvider); err != nil {
+		return nil, err
+	}
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"issuer":             storedNameProvider.Issuer,
+			"allowed_client_ids": storedNameProvider.AllowedClientIDs,
+			"scopes":             storedNameProvider.Scopes,
+		},
+	}, nil
+}
+
+// pathOIDCDeleteProvider is used to delete an assignment
+func (i *IdentityStore) pathOIDCDeleteProvider(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+	err := req.Storage.Delete(ctx, namedProviderPath+name)
+	if err != nil {
+		return nil, err
+	}
+	return nil, nil
```

Comment:
Nit: this can be simplified to `return req.Storage.Delete(ctx, namedProviderPath+name)`